torch/csrc/jit/codegen/cuda/kernel_ir_dispatch.h

```diff
@@ -136,6 +136,33 @@ class TORCH_CUDA_CU_API OptInDispatch : public OptOutDispatch {
   virtual void unhandled(Node* stmt) final;
 };
 
+// Base visitor class that visits all nodes in provided vector<Expr*>.
+//
+// Includes visiting through scopes like IfThenElse and ForLoop, and tracks them
+// in scopes_ and for_loops_.
+//
+// Makes a copy of exprs at exprs_ which could be used to modify and return.
+//
+// When traversing through ITE/FLs it will use a copy
+// of the provided expressions to make it safe to insert/delete nodes.
+//
+// Provides a simple base class to inherit from for typical kir passes
+class KirVisitor : public OptOutDispatch {
```

Comment:
nit: Do we need `Kir` in the name here? It's in the `kir` namespace. Maybe just `IrVisitor`?

Reply:
Yeah, I realized that in the next PR can rename it in a follow up.